ext/pdo_mysql/tests/pdo_mysql_dsn_containing_credentials.phpt

```diff
@@ -0,0 +1,28 @@
+--TEST--
+PDO_MYSQL: Pass credentials in dsn instead of constructor params
+--SKIPIF--
+<?php
+require_once(dirname(__FILE__) . DIRECTORY_SEPARATOR . 'skipif.inc');
+require_once(dirname(__FILE__) . DIRECTORY_SEPARATOR . 'mysql_pdo_test.inc');
+MySQLPDOTest::skip();
+?>
+--FILE--
+<?php
+	require_once(dirname(__FILE__) . DIRECTORY_SEPARATOR . 'mysql_pdo_test.inc');
+
+	class MySQLPDOTestDiscardCredentialParams extends MySQLPDOTest {
+		public function __construct($dsn, $user, $pass, $attr) {
+			return parent::__construct($dsn, null, null, $attr);
+		}
+	}
+
+	$link = MySQLPDOTestDiscardCredentialParams::factory('PDO', false, null, ['user' => PDO_MYSQL_TEST_USER, 'password' => PDO_MYSQL_TEST_PASS]);
+	echo "using credentials in dsn: done\n";
+
+	// test b/c - credentials in DSN are ignored when user/pass passed as separate params
+	$link = MySQLPDOTest::factory('PDO', false, null, ['user' => 'incorrect', 'password' => 'ignored']);
+	echo "ignoring credentials in dsn: done\n";
+?>
+--EXPECTF--
+using credentials in dsn: done
```

Comment:
I wouldn't mind adding a `SELECT USER()` but how would I test for `PDO_MYSQL_TEST_USER` in the EXPECT section? Currently the test would fail when the *incorrect* username is used btw.